packages/api/src/store/asset-table.ts

```diff
@@ -43,6 +44,17 @@ export default class AssetTable extends Table<WithID<Asset>> {
   }
 
   async getByIpfsCid(cid: string, user?: User): Promise<WithID<Asset>> {
+    let assetByStorageCid = await db.asset.getByStorageCid(cid, user);
+    let assetBySourceUrl = await db.asset.getBySourceURL("ipfs://" + cid, user);
```

Comment:
I think we should make this as a single round trip to the DB by making the condition on line `59` as an OR of either `storage->ipfs->>cid` or `source->>'url'`. We can probably keep the condition for only ready assets as well, no need to nest that condition only for the source URL check. WDYT?